src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -184,120 +193,152 @@ private static void CheckProtocolMethodParameters(ISymbolAnalysisContext context
             }
 
             context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0018, method.Locations.FirstOrDefault()), method);
         }
 
-        private static void CheckClientMethodReturnType(ISymbolAnalysisContext context, IMethodSymbol method)
+        private static bool IsOrImplements(ITypeSymbol typeSymbol, string typeName)
         {
-            bool IsOrImplements(ITypeSymbol typeSymbol, string typeName)
+            if (typeSymbol.Name == typeName)
             {
-                if (typeSymbol.Name == typeName)
-                {
-                    return true;
-                }
-
-                if (typeSymbol.BaseType != null)
-                {
-                    return IsOrImplements(typeSymbol.BaseType, typeName);
-                }
+                return true;
+            }
 
-                return false;
+            if (typeSymbol.BaseType != null)
+            {
+                return IsOrImplements(typeSymbol.BaseType, typeName);
             }
 
+            return false;
+        }
+
+        private static void CheckClientMethodReturnType(ISymbolAnalysisContext context, IMethodSymbol method)
+        {
             ITypeSymbol originalType = method.ReturnType;
             ITypeSymbol unwrappedType = method.ReturnType;
 
             if (method.ReturnType is INamedTypeSymbol namedTypeSymbol &&
                 namedTypeSymbol.IsGenericType &&
-                namedTypeSymbol.Name == "Task")
+                namedTypeSymbol.Name == TaskTypeName)
             {
                 unwrappedType = namedTypeSymbol.TypeArguments.Single();
             }
 
-            if (IsOrImplements(unwrappedType, "Response") ||
-                IsOrImplements(unwrappedType, "NullableResponse") ||
-                IsOrImplements(unwrappedType, "Operation") ||
-                IsOrImplements(originalType, "Pageable") ||
-                IsOrImplements(originalType, "AsyncPageable") ||
+            if (IsOrImplements(unwrappedType, ResponseTypeName) ||
+                IsOrImplements(unwrappedType, NullableResponseTypeName) ||
+                IsOrImplements(unwrappedType, OperationTypeName) ||
+                IsOrImplements(originalType, PageableTypeName) ||
+                IsOrImplements(originalType, AsyncPageableTypeName) ||
                 originalType.Name.EndsWith(ClientSuffix))
             {
                 return;
             }
 
             context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0015, method.Locations.FirstOrDefault(), originalType.ToDisplayString()), method);
-
         }
 
+        private bool IsCheckExempt(IMethodSymbol method)
+        {
+            return method.MethodKind != MethodKind.Ordinary ||
+                method.DeclaredAccessibility != Accessibility.Public ||
+                method.OverriddenMethod != null ||
+                method.IsImplicitlyDeclared ||
+                (method.Name.StartsWith("Get") && method.Name.EndsWith(ClientSuffix));
```

Comment:
There may be some subclients that don't have the `Client` suffix: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclient-factory-methods-suffix

Reply:
I didn't find one in azure-sdk-for-net. Do you have one?

Reply:
Oh sorry I missed that guideline link above. Then there do have defect here.